File: app/Http/Controllers/SpecialistPatternsJava/Creational/Builder/BuilderJava/src/Documentation.java
// Класс Documentation представляет объект документации и содержит информацию о создании документации.
public class Documentation {
     // Флаг для базовой документации
    private boolean base = false;
       // Флаг для состояния строительства
    private boolean building = false;
     // Флаг для дополнительных линий обслуживания
    private boolean serviceLines = false;
        // Флаг для завершения
    private boolean finish = false;

    // Геттер для флага базовой документации.
    public boolean isBase() {
        return base;
    }

    // Сеттер для флага базовой документации.
    public void setBase(boolean base) {
        this.base = base;
    }

    // Геттер для флага состояния строительства.
    public boolean isBuilding() {
        return building;
    }

    // Сеттер для флага состояния строительства.
    public void setBuilding(boolean building) {
        this.building = building;
    }

    // Геттер для флага дополнительных линий обслуживания.
    public boolean isServiceLines() {
        return serviceLines;
    }

    // Сеттер для флага дополнительных линий обслуживания.
    public void setServiceLines(boolean serviceLines) {
        this.serviceLines = serviceLines;
    }

    // Геттер для флага завершения.
    public boolean isFinish() {
        return finish;
    }

    // Сеттер для флага завершения.
    public void setFinish(boolean finish) {
        this.finish = finish;
    }

    // Метод YN возвращает "Да" или "Нет" в зависимости от значения boolean.
    public String YN(boolean r) {
        return r ? "Да" : "Нет";
    }

    // Переопределенный метод toString() возвращает информацию о состоянии документации.
    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("Базовая документация : ").append(YN(isBase())).append('\n')
          .append("Строительство : ").append(YN(isBuilding())).append('\n')
          .append("Дополнительные линии обслуживания : ").append(YN(isServiceLines())).append('\n')
          .append("Завершено : ").append(YN(isFinish())).append('\n');
        return sb.toString();
    }
}

File: app/Http/Controllers/SpecialistPatterns/Builder/Documentation.php
<?php

namespace App\Http\Controllers\SpecialistPatterns\Builder;

// Класс Documentation представляет объект документации и содержит информацию о создании документации.
use JetBrains\PhpStorm\Pure;

class Documentation {
    // Флаг для базовой документации
    private bool $base = false;
    // Флаг для состояния строительства
    private bool $building = false;
    // Флаг для дополнительных линий обслуживания
    private bool $serviceLines = false;
    // Флаг для завершения
    private bool $finish = false;

    // Геттер для флага базовой документации.
    public function isBase(): bool
    {
        return $this->base;
    }

    // Сеттер для флага базовой документации.
    public function setBase($base) {
        $this->base = $base;
    }

    // Геттер для флага состояния строительства.
    public function isBuilding(): bool
    {
        return $this->building;
    }

    // Сеттер для флага состояния строительства.
    public function setBuilding($building) {
        $this->building = $building;
    }

    // Геттер для флага дополнительных линий обслуживания.
    public function isServiceLines(): bool
    {
        return $this->serviceLines;
    }

    // Сеттер для флага дополнительных линий обслуживания.
    public function setServiceLines($serviceLines) {
        $this->serviceLines = $serviceLines;
    }

    // Геттер для флага завершения.
    public function isFinish(): bool
    {
        return $this->finish;
    }

    // Сеттер для флага завершения.
    public function setFinish($finish) {
        $this->finish = $finish;
    }

    // Метод YN возвращает "Да" или "Нет" в зависимости от значения boolean.
    public function YN($r): string
    {
        return $r ? "Да" : "Нет";
    }

    // Переопределенный метод __toString() возвращает информацию о состоянии документации.
    #[Pure] public function __toString() {
        $result = "Базовая документация : " . $this->YN($this->isBase()) . '<br />';
        $result .= "Строительство : " . $this->YN($this->isBuilding()) . '<br />';
        $result .= "Дополнительные линии обслуживания : " . $this->YN($this->isServiceLines()) . '<br />';
        $result .= "Завершено : " . $this->YN($this->isFinish()) . '<br />';

        return $result;
    }
}
